File: orange_hrm_ts/tests/steps/search.steps.ts
import { Page, expect } from '@playwright/test';
import { EmployeePage } from '../pages/EmployeePage';


export async function buscarFuncionarioPorNome(page: Page, name: string) {
  const employee = new EmployeePage(page);

  await employee.goToPIM();

  await employee.searchByName(name);

  await page.waitForSelector('.oxd-autocomplete-dropdown');
  await page.locator('.oxd-autocomplete-option').first().click();

  await employee.clickSearch();

  const resultRow = page
    .locator('.oxd-table-row')
    .filter({ hasText: name });

  await expect(resultRow).toHaveCount(1);
  await expect(resultRow.first()).toBeVisible();
  await expect(resultRow.first()).toContainText(name);
}

File: orange_hrm_python/tests/steps/search_steps.py
from playwright.sync_api import expect
from tests.pages.employee_page import EmployeePage

def buscar_funcionario(page, name):
    employee = EmployeePage(page)

    employee.go_to_pim()

    # campo autocomplete (robusto)
    field = page.locator(".oxd-form").get_by_role("textbox").first
    field.fill(name)

    page.wait_for_selector(".oxd-autocomplete-dropdown")
    page.locator(".oxd-autocomplete-option").first.click()

    page.get_by_role("button", name="Search").click()

    page.wait_for_timeout(1500)

    result = page.locator(".oxd-table-row").filter(has_text=name)

    expect(result.first).to_be_visible()